runtime/foucoco/src/lib.rs

```diff
@@ -962,254 +962,191 @@ where
+
+				warn!(
+					"Calling allowance() for currency {:?}, owner {:?} and spender {:?}",
+					currency_id, owner, spender
+				);
+
+				ensure!(
+					orml_currencies_allowance_extension::Pallet::<T>::is_allowed_currency(
 						currency_id,
-					);
+					),
+					DispatchError::Other("CurrencyId is not allowed for chain extension")
+				);
 
-				env.write(&total_supply.encode(), false, None).map_err(|_| {
-					DispatchError::Other("ChainExtension failed to call total_supply")
-				})?;
-			},
+				let allowance = orml_currencies_allowance_extension::Pallet::<T>::allowance(
+					currency_id,
+					&owner,
+					&spender,
+				);
 
-			//approve_transfer
-			1108 => {
+				env.write(&allowance.encode(), false, None)
+					.map_err(|_| DispatchError::Other("ChainExtension failed to call balance"))?;
+			},
+			// approve(currency, spender, amount)
+			1105 => {
 				let ext = env.ext();
-				let address = ext.address().clone();
 				let caller = ext.caller().clone();
 
-				let mut env = env.buf_in_buf_out();
+				let env = env.buf_in_buf_out();
 				let input = env.read(256)?;
-				let (origin_type, currency_id, to, amount): (
-					OriginType,
+				let (currency_id, spender, amount): (
 					CurrencyId,
 					T::AccountId,
 					BalanceOfForChainExt<T>,
 				) = chain_ext::decode(input)
 					.map_err(|_| DispatchError::Other("ChainExtension failed to decode input"))?;
 
-				let from = if origin_type == OriginType::Caller { caller } else { address };
-
-				warn!("from : {:#?}", from);
-				warn!("origin_type : {:#?}", origin_type);
-				warn!("to : {:#?}", to);
-				warn!("amount : {:#?}", amount);
+				warn!(
+					"Calling approve() allowing spender {:?} to transfer {:?} {:?} from {:?}",
+					spender, amount, currency_id, caller
+				);
 
-				let is_allowed_currency =
+				ensure!(
 					orml_currencies_allowance_extension::Pallet::<T>::is_allowed_currency(
 						currency_id,
-					);
-				if !is_allowed_currency {
-					return Err(DispatchError::Other(
-						"Currency id is not allowed for chain extension",
-					))
-				}
+					),
+					DispatchError::Other("CurrencyId is not allowed for chain extension",)
+				);
 
-				let result = orml_currencies_allowance_extension::Pallet::<T>::do_approve_transfer(
+				orml_currencies_allowance_extension::Pallet::<T>::do_approve_transfer(
 					currency_id,
-					&from,
-					&to,
+					&caller,
+					&spender,
 					amount,
-				);
-
-				warn!("approve_transfer : {:#?}", result);
-
-				match result {
-					DispatchResult::Ok(_) => {},
-					DispatchResult::Err(e) => {
-						let err =
-							Result::<(), ChainExtensionError>::Err(ChainExtensionError::from(e));
-						env.write(&err.encode(), false, None).map_err(|_| {
-							error!("ChainExtension failed to call 'approve'");
-							DispatchError::Other("ChainExtension failed to call 'approve'")
-						})?;
-					},
-				}
+				)?;
 			},
-
-			//transfer_approved
-			1109 => {
+			// transfer_from(currency, sender, recipient, amount)
```

Comment:
it is actually sender before currency

Reply:
Changed it in [this](https://github.com/pendulum-chain/pendulum/pull/248/commits/8a17e828fef9edf6916d171de319b7e51927f1ad) commit